packages/core/src/delegation.js

```diff
@@ -198,14 +228,15 @@ export class Delegation {
   /**
    * Attach a block to the delegation DAG so it would be included in the
    * block iterator.
-   * ⚠️ You should only attach blocks that are referenced from the `capabilities`
-   * or `facts`, if that is not the case you probably should reconsider.
-   * ⚠️ Once a delegation is de-serialized the attached blocks will not be re-attached.
+   * ⚠️ You can only attach blocks that are referenced from the `capabilities`
+   * or `facts`.
    *
    * @param {API.Block} block
    */
   attach(block) {
```

Comment:
Can we modify this to take `IPLDView` instead ? That way we can attach things that span multiple blocks and allow us to use this for `access/delagate` and `access/claim` apis

Reply:
I will add this as new PR given it is a breaking change.